sortinghat/parsing/grimoirelab.py

```diff
@@ -0,0 +1,355 @@
+                    identity = Identity(username=username, source=pb)
+                    my_ids.append(identity)
+
+            return my_ids
+
+
+        yaml = self.__load_yml(stream)
+
+        try:
+            for yid in yaml:
+                profile = yid['profile']
+                if profile is None:
+                    raise AttributeError('profile')
+
+                #we want the KeyError if name is missing
+                name = yid['profile']['name']
+                is_bot = profile.get('is_bot', False)
+
+                emails = yid.get('email', None)
+                enrollments = yid.get('enrollments', None)
+
+                first_email, first_username = self.__first_email_username(yid)
+                uuid = self.__compose_uuid(name, first_email, first_username)
+
+                uid = UniqueIdentity(uuid=uuid)
+
+                prf = Profile(name=name, is_bot=is_bot)
+                uid.profile = prf
+
+                # now it is time to add the identities for name, emails and backends
+                sh_identities = __create_sh_identities(name, emails, yid)
+                uid.identities += sh_identities
+
+                if enrollments:
+                    affiliations = self.__parse_affiliations_yml(enrollments, uuid)
+                    uid.enrollments += affiliations
+
+                self._identities[uuid] = uid
+
+        except KeyError as e:
+            msg = "invalid GrimoireLab yaml format. Attribute %s not found" % e.args
+            raise InvalidFormatError(cause=msg)
+
+    def __parse_organizations(self, stream):
+        """Parse GrimoireLab organizations.
+
+        The GrimoireLab organizations format is a YAML element stored
+        under the "organizations" key. The next example shows the
+        structure of the document:
+
+        - organizations:
+            Bitergia:
+                - bitergia.com
+                - support.bitergia.com
+                - biterg.io
+            LibreSoft:
+                - libresoft.es
+
+        :param json: YAML object to parse
+
+        :raises InvalidFormatError: raised when the format of the YAML is
+            not valid.
+        """
+        if not stream:
+            return
+
+        yaml = self.__load_yml(stream)
+
+        try:
+            for element in yaml:
+                name = self.__encode(element['organization'])
+
+                if not name:
+                    msg = "invalid GrimoireLab yaml format. Empty organization name"
+                    raise InvalidFormatError(cause=msg)
+
+                o = Organization(name=name)
+
+                if 'domains' in element:
+                    if isinstance(element['domains'], list):
+                        for dom in element['domains']:
+                            if dom:
+                                d = Domain(domain=dom, is_top_domain=False)
+                                o.domains.append(d)
+                            else:
```

Comment:
I think you can raise this error before so you can avoid one level of indentation. See my comment below.